libraries/ms-drivers/src/max6643_fan_controller.c

```diff
@@ -0,0 +1,35 @@
+#include <stddef.h>
+#include <string.h>
+
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "max6643_fan_controller.h"
+
+StatusCode max6643_init(Max6643Storage *storage, Max6643Settings *settings) {
+  if (storage == NULL || settings == NULL) {
+    return status_code(STATUS_CODE_INVALID_ARGS);
+  }
+
+  memset(storage, 0, sizeof(Max6643Storage));
+
+  storage->overtemp_pin = settings->overtemp_pin;
+  storage->fanfail_pin = settings->fanfail_pin;
+  storage->overtemp_callback = settings->overtemp_callback;
+  storage->fanfail_callback = settings->fanfail_callback;
+  storage->fanfail_callback_context = settings->fanfail_callback_context;
+  storage->overtemp_callback_context = settings->overtemp_callback_context;
+
+  static InterruptSettings s_interrupt_settings = {
+    .type = INTERRUPT_TYPE_INTERRUPT,
+    .priority = INTERRUPT_PRIORITY_NORMAL,
+  };
```

Comment:
It isn't good practice to declare static variables inside functions - that makes the value persist throughout calls to the function and it's very confusing. In this case you don't need it to take up memory for the whole program by making it static, so just make it a normal non-static variable. (You can make it `const` if you prefer.)